patches/tModLoader/Terraria/WorldGen.TML.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using Terraria.ModLoader;
+
+namespace Terraria
+{
+	public partial class WorldGen
+	{
+		public static WorldGenLocals GenLocalValues;
+		internal static event Action UpdateGenLocals;
+
+		internal static void InvokeGenLocalUpdate() {
+			UpdateGenLocals?.Invoke();
+		}
```

Comment:
Remove the `event` modifier. Then:
- You can use `UpdateGenLocals =` rather than `+=`
- You don't need to unsubscribe (`-=`)
- You don't need this `InvokeGenLocalUpdate` method